CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -691,13 +691,8 @@ function Patient:tickDay()
   -- It is nice to see plants, but dead plants make you unhappy
   self.world:findObjectNear(self, "plant", 2, function(x, y)
     local plant = self.world:getObject(x, y, "plant")
-  if not plant then
-    return
-  end
-    if plant:isPleasing() then
-      self:changeAttribute("happiness", 0.0002)
-    else
-      self:changeAttribute("happiness", -0.0002)
+    if plant then
+      self:changeAttribute("happiness", -0.0003 + (plant:isPleasingFactor() * 0.0001))
```

Comment:
This goes from `-0.0003` to `-0.0003 + 0.0001 = -0.0002`, ie always decreasing happiness.

Reply:
PleasingFactor is an integer 1 to 5

Reply:
Good point.

However, shouldn't the factor be interpolated then? I mean `self.phases - self.current_state` isn't hard-coded 1 to 5, `self.phases` is decided at runtime by parsing the animation. Change the animation, and the plant will have a different effect.

Reply:
Unless of course you decide that the length will always be 5, but in that case, it may need an assert or so to verify.

Reply:
Interpolation breaks the game if in multiplayer two players use different animations for the plant.

That seems to suggest it needs a fixed number of states, and the plant display should be interpolated?? (or just don't bother and decide 5).

Reply:
@mugmuggy could you fix phases at 5 (or go all the way with supporting a variable number of animations) as per Albert's suggestion.

Reply:
Sure I can change the initialisation back to just 5 for the self.phases for now. You have the same problem in a multiplayer game with everything else if any animation sequence is different, as the 2 states would be completely different, but where state doesn't necessarily need to sync, yes there would be issues.

Reply:
Yeah, we'd probably force base game animations to use the same number, and require multiplayer to have the same mods installed. Very out of scope.